ci/render_task.py

```diff
@@ -17,49 +17,58 @@
 
 def main():
     parser = argparse.ArgumentParser()
+    parser.add_argument('--use-secrets-server', action='store_true')
     parser.add_argument('--outfile', default='tasks.yaml')
+    parser.add_argument('--minimal', action='store_true', help='omit prebuild and promote steps')
```

Comment:
I'm not sure about the "minimal" build semantics tbh. there was the idea of adding "build targets", and to specify what the build does by adding or omitting those. I would prefer to stick to this approach (adding a flag on CLI-level as a convenience-shorthand seems okay-ish. but it should not be passed into the pipeline, i think